resolvers/manatee-primary-resolver/src/lib.rs

```diff
@@ -203,6 +210,49 @@ impl FromStr for ZkConnectString {
     }
 }
 
+//
+// Encapsulates a ZkConnectString with some bookkeeping to keep track of which
+// address the resolver should attempt to connect to next, and how many
+// connection attempts have failed in a row. Provides methods for getting the
+// next address, resetting the number of failed attempts, and checking if
+// the resolver should wait before trying to connect again.
+//
+#[derive(Debug, Clone)]
+struct ZkConnectStringState {
+    conn_str: ZkConnectString,
+    curr_idx: usize,
+    conn_attempts: usize,
+}
+
+impl ZkConnectStringState {
+    fn new(conn_str: ZkConnectString) -> Self {
+        ZkConnectStringState {
+            conn_str,
+            curr_idx: 0,
```

Comment:
Do you think it's worth randomizing which index we start with so that we spread our request load across all our zk instances or do you think it's not worth the extra complexity?

Reply:
That's a good idea! Done.